### PR TITLE
Feature/tasks/429719 implement services for producer fee calc

### PR DESCRIPTION
**Pull request by paulshimmen**

### CODE REVIEW

File: src/EPR.Payment.Facade.Common.UnitTests/RestServices/HttpRegistrationFeesServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+                HttpClientName = "HttpClientName"
+            };
+
+            _configMock = new Mock<IOptions<Service>>();
+            _configMock.Setup(x => x.Value).Returns(config);
+
+            _httpContextAccessorMock = new Mock<IHttpContextAccessor>();
+            _producerRegistrationFeesRequestDto = new ProducerRegistrationFeesRequestDto
+            {
+                ProducerType = "L",
```
Comment (manishagarwalEviden) [inline src/EPR.Payment.Facade.Common.UnitTests/RestServices/HttpRegistrationFeesServiceTests.cs:45]: isn't the producer type "large"? we are using 3 values - small, large or empty string

File: src/EPR.Payment.Facade.Common.UnitTests/RestServices/HttpRegistrationFeesServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+                           Content = new StringContent(JsonConvert.SerializeObject(_registrationFeesResponseDto), Encoding.UTF8, "application/json")
+                       });
+
+            var httpClient = new HttpClient(handlerMock.Object);
+            httpRegistrationFeesService = CreateHttpRegistrationFeesService(httpClient);
+
+            // Act
+            var result = await httpRegistrationFeesService.CalculateProducerFeesAsync(_producerRegistrationFeesRequestDto, cancellationToken);
+
+            // Assert
```
Comment (manishagarwalEviden) [inline src/EPR.Payment.Facade.Common.UnitTests/RestServices/HttpRegistrationFeesServiceTests.cs:157]: more than one assertion/verification. can be put in assertionscope

File: src/EPR.Payment.Facade.Common.UnitTests/RestServices/HttpRegistrationFeesServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+                       .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+                       .ThrowsAsync(new HttpRequestException("Unexpected error"));
+
+            var httpClient = new HttpClient(handlerMock.Object);
+            httpRegistrationFeesService = CreateHttpRegistrationFeesService(httpClient);
+
+            // Act
+            Func<Task> act = async () => await httpRegistrationFeesService.CalculateProducerFeesAsync(_producerRegistrationFeesRequestDto, cancellationToken);
+
+            // Assert
```
Comment (manishagarwalEviden) [inline src/EPR.Payment.Facade.Common.UnitTests/RestServices/HttpRegistrationFeesServiceTests.cs:184]: can be put in assertionscope

File: src/EPR.Payment.Facade.Common.UnitTests/RestServices/HttpRegistrationFeesServiceTests.cs [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+                       });
+
+            var httpClient = new HttpClient(handlerMock.Object);
+            httpRegistrationFeesService = CreateHttpRegistrationFeesService(httpClient);
+
+            // Act
+            Func<Task> act = async () => await httpRegistrationFeesService.CalculateProducerFeesAsync(_producerRegistrationFeesRequestDto, cancellationToken);
+
+            // Assert
+            await act.Should().ThrowAsync<ServiceException>()
```
Comment (manishagarwalEviden) [inline src/EPR.Payment.Facade.Common.UnitTests/RestServices/HttpRegistrationFeesServiceTests.cs:258]: could be put in assertionscope

File: src/EPR.Payment.Facade.Common/RESTServices/Payments/HttpPaymentsService.cs [hunk 5]
```diff
@@ -23,8 +20,6 @@ public HttpPaymentsService(
                 config.Value.EndPointName ?? throw new ArgumentNullException(nameof(config), ExceptionMessages.PaymentServiceEndPointNameMissing))
         {
             _httpContextAccessor = httpContextAccessor ?? throw new ArgumentNullException(nameof(httpContextAccessor));
```
Comment (manishagarwalEviden) [inline src/EPR.Payment.Facade.Common/RESTServices/Payments/HttpPaymentsService.cs:22]: I understand this is a clean up activity

**Review by manishagarwalEviden — CHANGES_REQUESTED** [review]

File: src/EPR.Payment.Facade.Common.UnitTests/RestServices/HttpRegistrationFeesServiceTests.cs [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+                .WithMessage($"{ExceptionMessages.RegistrationFeesServiceBaseUrlMissing} (Parameter 'config')");
+        }
+
+        [TestMethod, AutoMoqData]
+        public void Constructor_ConfigEndPointNameIsNull_ShouldThrowArgumentNullException(
+            Mock<IHttpContextAccessor> httpContextAccessorMock,
+            Mock<IHttpClientFactory> httpClientFactoryMock)
+        {
+            // Arrange
+            var configMock = new Mock<IOptions<Service>>();
```
Comment (manishagarwalEviden) [inline src/EPR.Payment.Facade.Common.UnitTests/RestServices/HttpRegistrationFeesServiceTests.cs:125]: configmock can be injected

**Review by manishagarwalEviden — CHANGES_REQUESTED** [review]

File: src/EPR.Payment.Facade.Common.UnitTests/RestServices/HttpRegistrationFeesServiceTests.cs (hunk 3 above)
Comment (paulshimmen) [reply src/EPR.Payment.Facade.Common.UnitTests/RestServices/HttpRegistrationFeesServiceTests.cs:184]: Sure np

File: src/EPR.Payment.Facade.Common.UnitTests/RestServices/HttpRegistrationFeesServiceTests.cs (hunk 4 above)
Comment (paulshimmen) [reply src/EPR.Payment.Facade.Common.UnitTests/RestServices/HttpRegistrationFeesServiceTests.cs:258]: Sure.

File: src/EPR.Payment.Facade.Common/RESTServices/Payments/HttpPaymentsService.cs (hunk 5 above)
Comment (paulshimmen) [reply src/EPR.Payment.Facade.Common/RESTServices/Payments/HttpPaymentsService.cs:22]: Yes, I noticed it was not needed in HttpRegistrationFeesService, so cleaned up HttpPaymentsService

**Review by manishagarwalEviden — APPROVED** [review]

**Pull request merged**